docs/design/build-system.md

```diff
@@ -0,0 +1,154 @@
+# Build System
+
+Angular CLI includes a first-party build system for Angular applications distributed as `@angular-devkit/build-angular`.
+This build system is responsible for creating a standalone application from user source files and third-party dependencies.
+
+`@angular-devkit/build-angular` itself integrates with the rest of Angular CLI by being an [Architect builder](https://angular.io/guide/cli-builder).
+This document describes a top level view of the functionality in `@angular-devkit/build-angular`, referred as just the "build system".
+Deprecated or soon to be removed features are not described here.
+
+In broad strokes the main areas are:
+- loading and processing sources
+- code splitting
+- production optimizations
+- post-processing steps
+
+Many tools are used in this process, and most of these steps happen within a [Webpack](https://webpack.js.org/) build.
+We maintain a number of Webpack-centric plugins in this repository, some of these are public but most are private since they are very specific to our setup.
+
+
+## Loading and processing sources
+
+Sources for Angular CLI browser apps are comprised of TypeScript files, style sheets, assets, scripts, and third party dependencies.
+A given build will load these sources from disk, process them, and bundle them together.
+
+
+### TypeScript
+
+Angular builds rely heavily on TypeScript-specific functionality for Ahead-of-Time template compilation (AOT).
+Outside Angular CLI, this is performed by the Angular Compiler (`ngc`), provided by `@angular/compiler-cli`.
```

Comment:
What exactly is the difference between `ngc` and `ngtsc`? Are they synonymous or do they refer to different binaries?

Reply:
IIRC `ngc` has always been the angular compiler, and `ngtsc` was the name sometimes given to the reworked `ngc` with Ivy. But that named was dropped. I don't think there's such a thing as `ngtsc` at the moment.

There were plans to have `ngc` be just a typescript plugin, but I think that's paused at the moment.

Reply:
I see. Probably makes sense to leave that out of the docs then, including it would only lead to more confusion for readers.